lib/membrane_rtc_engine/endpoints/hls/mobile_layout_maker.ex

```diff
@@ -0,0 +1,88 @@
+defmodule Membrane.RTC.Engine.Endpoint.HLS.MobileLayoutMaker do
+  @moduledoc """
+  Module representing function for updating video layout for the HLS stream.
+  """
+  @behaviour Membrane.RTC.Engine.Endpoint.HLS.VideoLayoutMaker
+
+  require Membrane.Pad
+
+  alias Membrane.Pad
+  alias Membrane.VideoCompositor.RustStructs.RawVideo
+  alias Membrane.VideoCompositor.RustStructs.VideoPlacement
+
+  @impl true
+  def init(output_caps), do: %{tracks: %{}, output_caps: output_caps}
+
+  @impl true
+  def track_added(state, %{metadata: %{"mainPresenter" => true}} = track) do
+    new_state = put_in(state, [:tracks, track.id], track)
+    updated_layout = [{Pad.ref(:input, track.id), get_track_layout(:main, 0, state.output_caps)}]
+
+    {updated_layout, new_state}
+  end
+
+  @impl true
+  def track_added(state, track) do
+    new_state = put_in(state, [:tracks, track.id], track)
+    {update_layout(new_state), new_state}
+  end
+
+  @impl true
+  def track_removed(state, track) do
+    {_, new_state} = pop_in(state, [:tracks, track.id])
+    {update_layout(new_state), new_state}
+  end
+
+  @spec update_layout(%{tracks: map(), output_caps: RawVideo.t()}) ::
+          list({Pad.ref_t(), VideoPlacement.t()})
+  defp update_layout(%{tracks: tracks, output_caps: output_caps}) do
+    tracks
+    |> Enum.filter(fn {_id, track} ->
+      track.type == :video and not track.metadata["mainPresenter"]
+    end)
+    |> Enum.with_index()
+    |> Enum.flat_map(fn {{_id, track}, index} ->
+      track_layout = get_track_layout(:basic, index, output_caps)
+      blank_layout = get_track_layout(:blank, index, output_caps)
+
+      [
+        {Pad.ref(:input, track.id), track_layout},
+        {Pad.ref(:input, {:blank, track.id}), blank_layout}
```

Comment:
I think we should convert blank frame/background into an option of `track_layout`.  I would wrap `VideoPlacement` into our struct that will have an additional field describing how the blank frame should look like e.g.

```elixir
defmodule VideoPlacement do
  @type t() :: %__MODULE__{
    video_placement: RustStructs.VideoPlacement.t(),
    options: [{:blank_frame?, boolean()}]
  }
end
```

In a more advanced version, instead of passing boolean as `blank_frame?` we could allow to specify frame width and height.

This way we will get rid of `pads` from our behavior and make everything clearer

Reply:
i think it is not necessary right now, it's only a temporary solution. I think that we can get back to this idea when there will be pull request to master. Then we'll know how the final version (release) of compositor api will look like.